Add time range handling:

JeanMarcMilletScality opened the pull request:
Add bars for every day in time range with value prefilled with 0 
Update tests


Next Step : Add unit Range

JeanMarcMilletScality committed:
Add time range handling:
Add bars for every day in time range with value prefilled with 0
Update tests

bert-e commented:
# Hello jeanmarcmilletscality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/core-ui/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @monpote

* @hervedombya

* @jeanmarcmilletscality

* @jbwatenbergscality

* @chengyanjin

ChengYanJin commented on src/lib/components/barchartv2/utils.ts:
```diff
@@ -36,6 +37,35 @@ export const getMaxValue = (data: { [key: string]: string | number }[]) => {
+ * @param endTimestamp - End timestamp in milliseconds
+ * @returns Array of timestamps for each day at midnight
+ */
+const generateDayTimestamps = (
```
Would be really great if we have unit test to convert this case:

it('should render when the starting data point is missing in the time range', async () => {})

ChengYanJin commented on src/lib/components/barchartv2/utils.ts:
```diff
@@ -36,6 +37,35 @@ export const getMaxValue = (data: { [key: string]: string | number }[]) => {
+  startDate.setHours(0, 0, 0, 0);
+  endDate.setHours(0, 0, 0, 0);
+
+  const currentDate = new Date(startDate);
```
Nit: just currentDate is bit confusing for me.
```suggestion
  const iteratingDate = new Date(startDate);
```

JeanMarcMilletScality committed:
ARTESCA-15250 Refactor Barchart component tests and enhance time data handling:
- Introduced interval handling for time ranges in Barchart.
- Updated tests to cover scenarios with missing data points and hourly intervals.
- Improved utility functions for generating time ranges and formatting timestamps.
- Added new storybook examples for time-based data with missing values.

ChengYanJin commented on src/lib/components/barchartv2/utils.ts:
```diff
@@ -39,61 +39,92 @@ export const getMaxValue = (data: { [key: string]: string | number }[]) => {
+    // Hourly intervals - use hour format
+    const hours = TIME_FORMATER.format(date);
+    const day = DAY_MONTH_FORMATER.format(date).replace(/[ ,]/g, '');
+    return `${day} ${hours}`;
```
For hourly intervals and minute intervals, maybe we could display the label only with hours. 
otherwise I feel too many repetition for day .

JeanMarcMilletScality committed:
Simplify timestamp formatting logic: for interval less than day, only show hour

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbwatenbergscality

* @monpote

* @chengyanjin

* @hervedombya

* @jeanmarcmilletscality

JeanMarcMilletScality commented:
/approve

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.0`




Please check the status of the associated issue ARTESCA-15250.

Goodbye jeanmarcmilletscality.




*The following options are set:* **approve**
